agent/consul/xdscapacity/capacity_test.go

```diff
@@ -0,0 +1,133 @@
+package xdscapacity
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/armon/go-metrics"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/time/rate"
+
+	"github.com/hashicorp/consul/acl"
+	"github.com/hashicorp/consul/agent/consul/state"
+	"github.com/hashicorp/consul/agent/structs"
+	"github.com/hashicorp/consul/sdk/testutil"
+)
+
+func TestController(t *testing.T) {
+	const index = 123
+
+	store := state.NewStateStore(nil)
+
+	for _, kind := range []structs.ServiceKind{
```

Comment:
It might be useful to add a short comment prior to this loop noting the desired outcome of the nested loops (100 proxy services + 25 non-proxy services).